src/Resistor/ResistorInput.tsx

```diff
@@ -0,0 +1,162 @@
+        <div className="ResistorDisplay">
+          <svg width="240" viewBox="0 0 136 40">
+            <defs>
+              <g id="Wire">
+                <rect y="20" height="1" width="8" fill="black" />
+              </g>
+              <g id="Resistor">
+                <rect
+                  y="8"
+                  x="8"
+                  width="120"
+                  height="24"
+                  rx="4"
+                  ry="4"
+                  stroke="black"
+                  strokeWidth="0.5"
+                  fill="burlywood"
+                />
+                <use xlinkHref="#Wire" x="0" />
+                <use xlinkHref="#Wire" x="128" />
+              </g>
+              <g id="Band">
+                <rect y="8" height="24" width="12" stroke="black" strokeWidth="0.5" />
+              </g>
+            </defs>
+            <use xlinkHref="#Resistor" x="0" y="0" />
+            {this.getBand(0)}
+            {this.getBand(1)}
+            {this.getBand(2)}
+            {this.getBand(3)}
+            {this.getBand(4)}
+            Sorry, your browser does not support inline SVG.
+          </svg>
+        </div>
+        <div className="ResistorValue">Value: {this.state.value}</div>
+        <div className="ResistorSelectors">
+          <ButtonGroup>
+            {this.valueSelector(0, 'First Band')}
+            {this.valueSelector(1, 'Second Band')}
+            {this.multiplierSelector(2, 'Third Band')}
+            {this.multiplierOrNullSelector(3, 'Fourth Band')}
+            {this.toleranceSelector(4, 'Tolerance Band')}
+          </ButtonGroup>
+        </div>
+      </div>
+    );
+  }
+
+  private getBand(index: number) {
+    const band = this.state.bands[index];
+    if (band != null) {
+      return (
+        <use 
+          xlinkHref="#Band"
+          x={this.getBandOffset(index)}
+          fill={band!.colorCode}
+        />
+      );
+    }
+    return '';
+  }
+
+  private getBandOffset(index: number) {
+      // The tolerance band is spaced further apart than the rest of the bands
+      if (index === (this.state.bands.length - 1)) {
+        return 104;
+      }
+      return (18 * index) + 20;
+  }
+
+  private valueSelector(index: number, title: String) {
+    return this.colorSelector(index, title, Resistor.colorTable.filter(color => color.hasValue()));
+  }
+
+  private multiplierSelector(index: number, title: String) {
+    return this.colorSelector(index, title, Resistor.colorTable);
+  }
+
+  private multiplierOrNullSelector(index: number, title: String) {
+    return this.colorSelector(index, title, [new Color('None', '', 0)].concat(Resistor.colorTable));
+  }
+
+  private toleranceSelector(index: number, title: String) {
+    // When puzzle-lib bug #49 is fixed, switch this to hasTolerance
+    return this.colorSelector(
+      index, title, 
+      Resistor.colorTable.filter(color => color.toleranceInPercent !== undefined));
+  }
+
+  // tslint:disable no-any

```

Comment:
I really would prefer that if we're going to disable tslint that we disable a single line only.  That's what I've done elsewhere.